scripts/operations/configuration/python_lib/product_catalog.py

```diff
@@ -208,3 +227,101 @@ def get_latest_csm_cfs_information(self) -> Dict[str, str]:
         CSM-specific wrapper for get_latest_cfs_information
         """
         return self.get_latest_cfs_information("csm")
+
+
+    def update_product_images_recipes(self, product_name: str, product_version: str,
+                                      image_name_ids: NameIdTupleList = None,
+                                      recipe_name_ids: NameIdTupleList = None) -> None:
+        """
+        Verifies that the specified product and version are in the product catalog.
+        Updates the 'images' and 'recipes' fields of the specified version of the specified product,
+        using the latest version of the cray-product-catalog-update tool.
+        """
+        installed_product_map = self.get_installed_products()
+        if product_name not in installed_product_map:
+            log_error_raise_exception(
+                f"update_product_images_recipes: Product '{product_name}' not in product catalog")
+        if product_version not in installed_product_map[product_name]:
+            log_error_raise_exception(f"update_product_images_recipes: Version '{product_version}'"
+                                      f" of product '{product_name}' not in product catalog")
+        latest_cpc_update_version = get_latest_cpc_update_version()
+        content_string_list = []
+        if image_name_ids:
+            image_name_ids_content_string_list = [ '"%s": {id: "%s"}' % (image_name, image_id)
+                                                   for image_name, image_id in image_name_ids ]
+            image_name_ids_content_string = ", ".join(image_name_ids_content_string_list)
+            content_string_list.append("images: {%s}" % image_name_ids_content_string)
+        if recipe_name_ids:
+            recipe_name_ids_content_string_list = [ '"%s": {id: "%s"}' % (recipe_name, recipe_id)
+                                                    for recipe_name, recipe_id in recipe_name_ids ]
+            recipe_name_ids_content_string = ", ".join(recipe_name_ids_content_string_list)
+            content_string_list.append("recipes: {%s}" % recipe_name_ids_content_string)
+        yaml_content_string = "{%s}" % ", ".join(content_string_list)
```

Comment:
I mentioned this to Mitch already in a slack message, but I'm wondering if we could use PyYAML to create the YAML string from a Python object instead, so we don't have to construct the YAML string manually. It's a little harder to follow than it needs to be.

Reply:
Fixed.